compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala

```diff
@@ -45,6 +46,18 @@ class InterceptedMethods extends MiniPhase {
       ctx.log(s"$phaseName rewrote $tree to $rewritten")
       rewritten
     }
+    else if (tree.name.startsWith(nme.UNARY_PREFIX.toString) && tree.qualifier.symbol.isAbsent) {
+      tree.qualifier match {
+        case refTree: RefTree => tree
```

Comment:
I believe this is dead code otherwise `tree.qualifier.symbol.isAbsent` would not be `true`